src/Lean/Compiler/LCNF/JoinPoints.lean

```diff
@@ -5,11 +5,94 @@ Authors: Henrik Böving
 -/
 import Lean.Compiler.LCNF.CompilerM
 import Lean.Compiler.LCNF.PassManager
+import Lean.Compiler.LCNF.PullFunDecls
 
 namespace Lean.Compiler.LCNF
 
+-- TODO: These can be used in a much more general context
+partial def mapFVarM [Monad m] (f : FVarId → m FVarId) (e : Expr) : m Expr := do
+  match e with
+  | .proj typ idx struct => return .proj typ idx (←mapFVarM f struct)
```

Comment:
Small nitpicking: could you please use a space between the arrow and `mapFVarM`?